packages/block-library/src/post-comment-reply-link/index.php

```diff
@@ -0,0 +1,72 @@
+<?php
+/**
+ * Server-side rendering of the `core/post-comment-reply-link` block.
+ *
+ * @package WordPress
+ */
+
+/**
+ * Renders the `core/post-comment-reply-link` block on the server.
+ *
+ * @param array    $attributes Block attributes.
+ * @param string   $content    Block default content.
+ * @param WP_Block $block      Block instance.
+ * @return string Return the post comment's reply link.
+ */
+function render_block_core_post_comment_reply_link( $attributes, $content, $block ) {
+	if ( ! isset( $block->context['commentId'] ) ) {
+		return '';
+	}
+
+	$thread_comments = get_option( 'thread_comments' );
+	if ( ! $thread_comments ) {
+		return '';
+	}
+
+	$comment = get_comment( $block->context['commentId'] );
+
+	$depth     = 1;
+	$max_depth = get_option( 'thread_comments_depth' );
+	$parent_id = $comment->comment_parent;
+
+	// Compute comment's depth iterating over its ancestors.
+	while ( 0 !== $parent_id ) {
```

Comment:
@aristath, can you think of a better way to compute the depth of the current comment?

The only concern I have here is that I would feel better if the check for the `$parent_id` wasn't so strict about it being a number. I would feel more comfortable seeing the loop terminated as soon as `$parent_id` has falsy value.
```suggestion
	while ( ! empty( $parent_id ) ) {
```

Reply:
I run into the issue like that:

<img width="1754" alt="Screen Shot 2021-10-26 at 17 01 26" src="https://user-images.githubusercontent.com/699132/138906456-eb1f5b18-6f69-49cf-b640-67074ea84125.png">

Reply:
It happens in two cases, when there is a comment selected with an invalid id, or when a comment doesn't have any parents.